editor/plugins/polygon_2d_editor_plugin.cpp

```diff
@@ -52,38 +53,37 @@
 #include "scene/gui/texture_rect.h"
 #include "scene/gui/view_panner.h"
 
-class UVEditDialog : public AcceptDialog {
-	GDCLASS(UVEditDialog, AcceptDialog);
-
-	void shortcut_input(const Ref<InputEvent> &p_event) override {
-		const Ref<InputEventKey> k = p_event;
-		if (k.is_valid() && k->is_pressed()) {
-			bool handled = false;
-
-			if (ED_IS_SHORTCUT("ui_undo", p_event)) {
-				EditorNode::get_singleton()->undo();
-				handled = true;
-			}
-
-			if (ED_IS_SHORTCUT("ui_redo", p_event)) {
-				EditorNode::get_singleton()->redo();
-				handled = true;
-			}
-
-			if (handled) {
-				set_input_as_handled();
-			}
-		}
-	}
-};
-
 Node2D *Polygon2DEditor::_get_node() const {
 	return node;
 }
 
 void Polygon2DEditor::_set_node(Node *p_polygon) {
+	CanvasItem *draw = Object::cast_to<CanvasItem>(canvas);
+	if (node) {
+		node->disconnect("draw", callable_mp(draw, &CanvasItem::queue_redraw));
+		node->disconnect("draw", callable_mp(this, &Polygon2DEditor::_update_available_modes));
+	}
+
 	node = Object::cast_to<Polygon2D>(p_polygon);
 	_update_polygon_editing_state();
+	canvas->queue_redraw();
+	if (node) {
+		canvas->set_texture_filter(node->get_texture_filter_in_tree());
+
+		if (!polygon_edit->is_inside_tree()) {
+			EditorNode::get_bottom_panel()->add_item(TTR("Polygon"), polygon_edit);
+		}
+		EditorNode::get_bottom_panel()->make_item_visible(polygon_edit);
+		_update_bone_list();
+		_update_available_modes();
+		if (current_mode == MODE_MAX) {
+			_select_mode(MODE_POINTS); // Initialize when opening the first time.
+		}
+		get_tree()->connect("process_frame", callable_mp(this, &Polygon2DEditor::_center_view), CONNECT_ONE_SHOT);
```

Comment:
There has to be a better signal to use than that...
E.g. some `visibility_changed` with CONNECT_DEFERRED.

Reply:
I don't know, this keeps stumping me. The canvas is redrawn multiple times when opening it, and often first time with wrong size. Hence simply centering it upon opening/visibility change/first draw and whatever I have tested, even with deferred call, fails. I haven't found any other reliable way to center the view when first opening the editor other than waiting for the next frame.
I remember I really struggled with this also when I first introduced the feature in https://github.com/godotengine/godot/pull/83731